### PR TITLE
Fix XML framing and bound streaming input

### PR DESCRIPTION
**Pull request by robpelf**
## Summary

- preserve exact first-frame boundaries and coalesced response tails across the protocol reader and Unix/SSH clients
- apply the 64 MiB default byte limit and 256-element nesting limit independently to each XML frame, with additive configuration APIs
- reject malformed XML 1.0 constructs early, including invalid declarations/names/references, DOCTYPEs, reserved PI targets, literal `<` in attributes, and `]]>` in character data
- avoid repeated rescanning of incomplete tags, declarations, comments, CDATA, and references by tracking linear parser progress across chunks
- process fragmented and coalesced commands on mock Unix/TCP/SSH sessions; malformed or oversized requests receive a generic 400 response and close only the affected session
- reset buffered tails on reconnect and invalidate a client connection after a framing/protocol error

## Compatibility and scope

The public API changes are additive. Existing constructors retain a 64 MiB per-frame default, while callers can select a custom or disabled byte limit. XML depth remains bounded at 256 elements by default.

This PR addresses XML framing and bounded transport input. It does not add application-level report streaming or change GMP command/response models.

Refs #172

## Public evidence

Implementation and validation used only public sources:

- rust-gvm `devel` at `03a0e38118c8ccab538dad145e52eb5163b801b4`
- python-gvm at `2bb100fd03f02e598f046e2032e12550b5b14751`
- the public XML 1.0 grammar and quick-xml behavior represented by the repository's locked `quick-xml` 0.41.0 dependency

Transport behavior was exercised through real Unix sockets and SSH channels against the public `gvm-mock-server`, plus the pinned public python-gvm client lifecycle.

## Validation

- `cargo fmt --all -- --check`
- `cargo test --workspace --all-features`
- `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- `RUSTDOCFLAGS=-Dwarnings cargo doc --workspace --all-features --no-deps`
- `cargo +1.85.0 test --workspace`
- `cargo +1.85.0 check --workspace --all-features`
- pinned public python-gvm Unix integration: all 15 steps passed
- `cargo llvm-cov --workspace --all-features --lcov --output-path lcov.info`
- `cargo deny check`
- `cargo audit`
- `cargo machete`
- `cargo vet`
- `python3 scripts/check_workspace_unsafe.py`
- Semgrep `p/rust`, `p/security-audit`, and `p/secrets`: 0 findings
- `git diff --check`

A randomized fuzz campaign was not run on the final head. The streaming harness compiled earlier in the implementation slice; final-head deterministic regressions cover exact frame/tail reconstruction, single- and multi-chunk malformed input, and linear progress across a long incomplete attribute.

Known baseline warnings: cargo-deny reports unmatched allowances, cargo-audit reports the existing allowed yanked transitive `crypto-bigint 0.7.4` warning through `russh`, and MSRV tests emit pre-existing missing-docs warnings in integration crates. Cargo-vet produced only its known `imports.lock` quote-normalization churn, which was inspected and restored.

### CODE REVIEW

**Review by clawosiris — APPROVED** [review]
I did not find any blocking issues. The XML framing changes preserve exact frame boundaries, cover coalesced/tail handling, malformed XML, limits, and transport integration, and the full CI/security matrix is green. Residual risk is mostly in edge-case XML grammar behavior beyond the new tests, but this is a solid improvement.

File: crates/gvm-protocol/src/xml_reader.rs [hunk 1]
```diff
@@ -3,16 +3,265 @@
...
+            text,
+            *self.seen_start,
+            self.declaration_allowed,
+            self.trailing_text_brackets,
+        )? {
+            return Ok(false);
+        }
+        *self.resume_offset += parsed_len;
+        Ok(true)
+    }
```
[the remark this answers is not in the document]
Comment (clawosiris) [reply crates/gvm-protocol/src/xml_reader.rs:214]: No inversion is present. validate_text returns false for complete text, causing accept_text to return Ok(false), and check_complete therefore continues parsing. True is reserved for an incomplete UTF-8 tail, where early suspension after advancing the resume offset is intentional and covered by test_incomplete_multibyte_text_across_chunks. No code change is needed.

Agent: Thoth
